redis/redis.go

```diff
@@ -847,6 +860,45 @@ func (c *DefaultConn) MemoryStats() (map[string]any, error) {
 	return result, nil
 }
 
+func (c *DefaultConn) ReceiveStrings(n int) ([]string, error) {
+	replies := make([]string, 0, n)
+	err := c.receive(n, func(reply any, err error) error {
+		if err != nil {
+			return err
+		}
+		val, err := redis.String(reply, nil)
+		if errors.Is(err, redis.ErrNil) {
+			replies = append(replies, "")
+			return nil
+		}
+		if err != nil {
+			return err
```

Comment:
How to errors affect the pipelining? If any command in the pipe has an error will it ruin the whole batch?  Will it ruin the count that `c.receive` depends on?

Reply:
This is a callback function to c.receive. It doesn't modify the `count`, `c.receive` depends on.

The error here means something is wrong with redis and I'm think it's safer to bail on this whole batch then continue.

Reply:
I am ok bailing on the whole batch.